src/test/javascript/spec/component/text-assessment-dashboard/text-assessment-dashboard.component.spec.ts

```diff
@@ -0,0 +1,253 @@
+        expect(component.examId).toBe(2);
+    });
+
+    it('should get submissions', fakeAsync(() => {
+        // test getSubmissions
+        const exerciseServiceFind = jest.spyOn(exerciseService, 'find');
+        exerciseServiceFind.mockReturnValue(of(new HttpResponse({ body: textExercise })));
+        const getTextSubmissionStub = jest.spyOn(textSubmissionService, 'getTextSubmissionsForExerciseByCorrectionRound');
+        getTextSubmissionStub.mockReturnValue(of(new HttpResponse({ body: [textSubmission], headers: new HttpHeaders() })));
+        const isAtLeastInstructorInCourseStub = jest.spyOn(accountService, 'isAtLeastInstructorInCourse');
+        isAtLeastInstructorInCourseStub.mockReturnValue(true);
+
+        // call
+        component.ngOnInit();
+        tick(100);
+        // check
+        expect(getTextSubmissionStub).toHaveBeenCalledWith(textExercise.id, { submittedOnly: true });
+        expect(component.submissions).toEqual([textSubmission]);
+        expect(component.filteredSubmissions).toEqual([textSubmission]);
+    }));
+
+    it('should not get Submissions', () => {
+        const getTextSubmissionStub = jest.spyOn(textSubmissionService, 'getTextSubmissionsForExerciseByCorrectionRound');
+        getTextSubmissionStub.mockReturnValue(of(new HttpResponse({ body: [], headers: new HttpHeaders() })));
+        const isAtLeastInstructorInCourseStub = jest.spyOn(accountService, 'isAtLeastInstructorInCourse');
+        isAtLeastInstructorInCourseStub.mockReturnValue(true);
+        const findExerciseStub = jest.spyOn(exerciseService, 'find');
+        findExerciseStub.mockReturnValue(of(new HttpResponse({ body: textExercise, headers: new HttpHeaders() })));
+        component.exercise = textExercise;
+        // call
+        component.ngOnInit();
+
+        // check
+        expect(findExerciseStub).toHaveBeenCalled();
+        expect(getTextSubmissionStub).toHaveBeenCalledWith(textExercise.id, { submittedOnly: true });
+        expect(component.submissions).toEqual([]);
+        expect(component.filteredSubmissions).toEqual([]);
+    });
+
+    it('should update filtered submissions', () => {
+        // setup
+        component.ngOnInit();
+        component.updateFilteredSubmissions([textSubmission]);
+        // check
+        expect(component.filteredSubmissions).toEqual([textSubmission]);
+    });
+
+    it('should cancel assessment', fakeAsync(() => {
+        // test cancelAssessment
+        const windowSpy = jest.spyOn(window, 'confirm').mockReturnValue(true);
+        const cancelAssessmentStub = jest.spyOn(textAssessmentService, 'cancelAssessment').mockReturnValue(of());
+        component.exercise = textExercise;
+
+        // call
+        component.cancelAssessment(textSubmission);
+        tick();
+
+        // check
+        expect(cancelAssessmentStub).toHaveBeenCalledWith(textSubmission.participation.id, textSubmission.id);
+        expect(windowSpy).toHaveBeenCalled();
```

Comment:
```suggestion
        expect(windowSpy).toHaveBeenCalledTimes(x);
```
Can we define a number of times the spy has been called?